src/contexts/User-Context/user-context.js

```diff
@@ -125,6 +125,26 @@ const AuthProvider = ({ children }) => {
     }
   };
 
+  const getUserWatchLater = async () => {
+    if (!userState.isLoggedIn || !userDetails) {
+      return;
+    }
+    try {
+      const {
+        data: { watchlater },
+      } = await callAPI(
+        "GET",
+        "/api/user/watchlater",
+        null,
+        userDetails?.encodedToken
+      );
+      console.log(watchlater);
+      userDispatch({ type: "WATCH_LATER", payload: watchlater });
+    } catch (error) {
+      toast.error("Problem in retrieving your watch later videos!");
```

Comment:
Suggestion: You can also display the error message directly from the api like this so that user would know the exact error ` toast.error(e?.response?.data?.errors[0]);`

Reply:
Yeah sure will do that